### ISSUE TITLE
`no-collection-bracket-access`: false positive on dynamic method call

### ISSUE THREAD

**Issue by silverwind** [sindresorhus/eslint-plugin-unicorn#3461]
`no-collection-bracket-access` flags computed bracket access even when the key is a collection method (here used dynamically):

```js
const set = new Set();
set[set.has('x') ? 'delete' : 'add']('x'); // should be ok
```

Suggesting to never flag bracket notation accessing collection methods like `add`, `delete` etc, which means `set["add"]` likely should be valid.
